### ISSUE TITLE
TbTb under Windows

### ISSUE THREAD

**Issue by DavidBrainard** [ToolboxHub/ToolboxToolbox#73]
My impression is that some people out there are using TbTb under Windows.  We tried to set this up for a visitor yesterday and got an error when TbTb tried to figure out how to invoke git.  TbTb issued a "which ...." command, which works under Mac/Linux, but this failed under Windows.

As a work around, we just commented out that bit of code and hard coded in the command as "git", which then appeared to work just fine on the particular system.

If others have seen this problem and found a better fix, I'd like to fold that into the distribution with a conditional on the type of machine being run.

**Comment by DavidBrainard** [comment]
Here are the steps we used to get TbTb to work on a computer.

First, we verified that git was installed and working,  We did this by executing
  system('git') 
from the command window and watching some output stream by, and not
getting an error.

Then we had to fix two three things.

a) There is a place where TbTb checks for the existence of git.  We had to comment that out and assert that it works. 

b) There is a place where TbTb tries to determine the exact git command to execute.  We had to comment that out and just say that 'git' would work.

c) There were some spaces in the path to the userpath folder.  Windows and TbTb do not play well with those.  We had to change those.  It is possible there would be some way to quote file paths in TbTb so that these worked.

I will try to fix a) and b) with machine conditionals when I get a chance.

**Comment by ragavsathish** [comment]
Professor @DavidBrainard , 
I am working with TbTb on windows system, I will fix this and send you the pull request asap. 
Regards,
Sathish

**Comment by DavidBrainard** [comment]
Thanks!  Windows is a pretty foreign land for me.

**Comment by DavidBrainard** [comment]
These issues addressed by ragavsathish.  Closing, pending any reports that things have broken somewhere as a result.

**Comment by ragavsathish** [comment]
I did run the test `tbAssertTestsPass`. As am using windows 7 with no subversion, except for docker & svn test everything else is passed in windows pc. Not sure about linux machine

**Comment by DavidBrainard** [comment]
There was one little issue on Mac/Linux.  In the tests directory, the tempdir got modified to test how things worked when there was a space in the pathname.  Turns out that white space in paths does not actually work on Mac/Linux, at least in one instances probed by the tests.  It is possible that we never had a case with white space in the name that we were trying to deal with on Mac/Linux, or that the test itself probes a case we never really care about.

I fixed this by making the addition of white space in the test conditional on ispc, and the tests now pass on Mac/Linux. 

I don't think that will have any negative impact on the Windows side, but if it does let me know and we'll figure it out.

Thanks for this fix!

Closing this issue.  Can reopen if there is something hanging.
